File: resQfood-user_management/src/main/java/models/ali/Basket.java
package models.ali;

import java.sql.Date;

public class Basket {
    // Attributes
    private final int basket_id;
    private String basket_status;
    private final int user_id;
    private java.sql.Date confirmation_date; // Replaced event_id with confirmation_date

    // Constructors
    public Basket(int basket_id, String basket_status, int user_id, Date confirmation_date) {
        this.basket_id = basket_id;
        this.basket_status = basket_status;
        this.user_id = user_id;
        this.confirmation_date = confirmation_date;
    }

    // Getter methods
    public int getBasketId() {
        return basket_id;
    }

    public String getBasketStatus() {
        return basket_status;
    }

    public int getUserId() {
        return user_id;
    }

    public Date getConfirmationDate() {
        return confirmation_date;
    }

    // Setter methods (if needed)
    public void setBasketStatus(String basket_status) {
        this.basket_status = basket_status;
    }

    public void setConfirmationDate(Date confirmation_date) {
        this.confirmation_date = confirmation_date;
    }

    // Additional methods (if needed)

    @Override
    public String toString() {
        return "Basket [basket_id=" + basket_id + ", basket_status=" + basket_status +
                ", user_id=" + user_id + ", confirmation_date=" + confirmation_date + "]";
    }
}

File: src/Entity/Basket.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

/**
 * Basket
 *
 * @ORM\Table(name="basket")
 * @ORM\Entity
 */
class Basket
{
    /**
     * @var int
     *
     * @ORM\Column(name="basket_id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $basketId;

    /**
     * @var string|null
     *
     * @ORM\Column(name="basket_status", type="string", length=100, nullable=true)
     */
    private $basketStatus;

    /**
     * @var int
     *
     * @ORM\Column(name="user_id", type="integer", nullable=false)
     */
    private $userId;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="confirmation_date", type="date", nullable=true, options={"default"="CURRENT_TIMESTAMP"})
     */
    private $confirmationDate = 'CURRENT_TIMESTAMP';

    public function getBasketId(): ?int
    {
        return $this->basketId;
    }

    public function getBasketStatus(): ?string
    {
        return $this->basketStatus;
    }

    public function setBasketStatus(?string $basketStatus): static
    {
        $this->basketStatus = $basketStatus;

        return $this;
    }

    public function getUserId(): ?int
    {
        return $this->userId;
    }

    public function setUserId(int $userId): static
    {
        $this->userId = $userId;

        return $this;
    }

    public function getConfirmationDate(): ?\DateTimeInterface
    {
        return $this->confirmationDate;
    }

    public function setConfirmationDate(?\DateTimeInterface $confirmationDate): static
    {
        $this->confirmationDate = $confirmationDate;

        return $this;
    }


}
